Management API Rate Limits for Self Service Subscribers

tamigoodall opened the pull request:
Updated Management API Rate Limits and Entity Limits

Updated rate-limits.md - separating out Mgmt API and Auth API endpoint rate limit docs into separate, linked files
Added entity-limits.md
Added legacy-rate-limits.md

Reviews:
- https://docs-content-staging-pr-8982.herokuapp.com/docs/policies/rate-limits
- https://docs-content-staging-pr-8982.herokuapp.com/docs/policies/entity-limits
- https://docs-content-staging-pr-8982.herokuapp.com/docs/policies/legacy-rate-limits
- https://docs-content-staging-pr-8982.herokuapp.com/docs/policies/rate-limits-mgmt-api
- https://docs-content-staging-pr-8982.herokuapp.com/docs/policies/rate-limits-auth-api

<!---
Notes:
- Your PR should conform to our [Contributing Guidelines](https://github.com/auth0/docs/blob/master/CONTRIBUTING.md)
- If applicable, add details to the [update feed](https://github.com/auth0/docs/tree/master/updates)
- Make sure your PR gets deployed to a Heroku [Review App](https://github.com/auth0/docs/blob/master/CONTRIBUTING.md#review-apps) without errors
- Please include a short description
- If your PR is a work in progress, title it [DO NOT MERGE]
--->

tamigoodall committed:
Management API Rate Limits for Self Service Subscribers

Added new limits to rate-limits.md
Minor H2, H3, H4 header capitalization changes
Deleted old table under Management API v2

tamigoodall committed:
Added legacy and entity rate limits pages

Added content for legacy and entity rate limits pages
Added screenshot for limit reached

tamigoodall committed:
Update rate-limits.md

Added link to screenshot

tamigoodall committed:
Update legacy-rate-limits.md

Added legacy rate limit content

tamigoodall committed:
Links and minor edits

Links and minor edits

tamigoodall committed:
Sidebar entries

Sidebar entries
Minor title changes in index.md

tamigoodall committed:
Update rate-limits.md

Moved screenshot

tamigoodall committed:
Update sidebar.yml

Added hidden children: legacy-rate-limits and entity-limits

tamigoodall committed:
Separated rate limits by api

Separated rate limits by api

tamigoodall committed:
Added Entity Limits information

Replaced "self-service plan" where appropriate with "Developer and Developer Pro plans"
Added description of entity limits
Merged global-limit.md into rate-limits-auth-api.md contents
Added Keep reading links
Make titles consistent
Updated sidebar
Added screenshot for entity-limit-reached.png

tamigoodall committed:
Minor edits

Typos fixed and redundant screenshot removed

tamigoodall committed:
Update rate-limits.md

Clarified HTTP response header text

tamigoodall committed:
Update rate-limits-mgmt-api.md

Fixed table column alignment issues
Clarified text

tamigoodall committed:
Fixed date formats

Fixed date formats

tamigoodall committed:
Update rate-limits.md

Moved link to entity limits

tamigoodall committed:
Update rate-limits-mgmt-api.md

Fixed warning

tamigoodall committed:
Update rate-limits-mgmt-api.md

typo

tamigoodall committed:
Update rate-limits-mgmt-api.md

Standardized tables

tamigoodall committed:
Update rate-limits-mgmt-api.md

missing `

tamigoodall committed:
Fixed dates

Fixed dates

tamigoodall committed:
Changed dates to June 18, 2020

Changed dates to June 18, 2020

santiagoaguiar commented on articles/policies/entity-limits.md:
```diff
@@ -0,0 +1,46 @@
+
+Entities in Auth0 are tenant configuration elements such as applications, connections, rules, and API resource servers. 
+
+Auth0 limits the number of entities you can have depending on your subscription level. Auth0 provides notifications to you when you are approaching (90%) and when you have reached (100% or higher) your respective entity limits. We will also provide informative messages to prevent you from attempting to configure entities that would end up being rejected.
```
```suggestion
Auth0 limits the number of entities you can have depending on your subscription level. Auth0 provides notifications to you when you are approaching (80%) and when you have reached (100% or higher) your respective entity limits. We will also provide informative messages to prevent you from attempting to configure entities that would end up being rejected.
```

tamigoodall replied:
manually updated with a separate commit

santiagoaguiar commented on articles/policies/entity-limits.md:
```diff
@@ -0,0 +1,46 @@
+
+![Entity Limit Reached](/media/articles/policies/entity-limit-reached.png)
+
+## Enterpise, Developer and Developer Pro subscription limits
```
We might want to mention that entity limits counts might take some seconds to be updated. If customer is still seeing a warning when they believe they shouldn't, they should try again after some seconds or contact support if the issue persists.

tamigoodall replied:
Added a note

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,97 @@
+| Rotate signing keys | `POST /api/v2/keys/signing/rotate` | n/a | n/a | 5 |
+| Create tenant | `POST /api/v2/tenants` | 5 | n/a | n/a |
+| Read connection status | `GET /api/v2/connections/{id}/status` | 15 | n/a | n/a |
+| Create tenant invitation | `POST /api/v2/tenants/invitations` | n/a | 5 | n/a |
```
This applies for everyone, not only self-service. Move to the table above.

tamigoodall replied:
Merged tables

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,97 @@
+| Read connections | `GET /api/v2/connections` | 10 | 100 |
+| | `POST /api/v2/connections/{id}` | |
+| Delete device credentials | `DELETE /api/v2/device-credentials/{id}` | 5 | 100 |
+| All other endpoints | | 10 | 150 |
```
This is only valid for self-service and is already described below, please remove this (also the affected endpoints are not needed).

tamigoodall replied:
I received some conflicting comments on these tables. I've tried to make the changes as described, but I may need you to review it again.

panga replied:
Np, I will review again.

panga commented on articles/policies/entity-limits.md:
```diff
@@ -0,0 +1,46 @@
+
+![Entity Limit Reached](/media/articles/policies/entity-limit-reached.png)
+
+## Enterpise, Developer and Developer Pro subscription limits
```
```suggestion
## Developer and Developer Pro subscription limits
```

Please make it clear that Enterprise subscriptions do not have limits.

tamigoodall committed:
Minor fixes

Clarified warning about dates
Fixed date format in title

tamigoodall committed:
Randy's review comments

Randy's review comments

tamigoodall committed:
Update rate-limits-mgmt-api.md

Merged table info
Deleted unneeded H2s
Added item for per day limit on rotate signing keys

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,99 @@
+| Read connections | `GET /api/v2/connections` | 10 | 100 |
+| | `POST /api/v2/connections/{id}` | |
+| Delete device credentials | `DELETE /api/v2/device-credentials/{id}` | 5 | 100 |
+| All other endpoints | | 10 | 150 |
```
Please remove this section.

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,99 @@
+| Endpoint | Path | Rate Limit (per second) | Rate Limit (per minute) | Rate Limit (per day) |
+| - | - | - | - | - |
+| Verify custom domain | `POST /api/v2/custom-domains{id}/verify` | n/a | 5 | n/a |
+| Send test email | `POST /api/v2/email-send-test` | 1 | n/a | n/a |
```
```suggestion
```
Not public. Please remove.

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,89 @@
+| Write device credentials | `POST /api/v2/device-credentials` | 5 | 100 | 
+| Delete device credentials | `DELETE /api/v2/device-credentials/{id}` | 5 | 100 |
+| Verify custom domain | `POST /api/v2/custom-domains{id}/verify` | n/a | 5 | 
+| Send test email | `POST /api/v2/email-send-test` | 1 | n/a | 
```
```suggestion
```
Not public. Please remove.

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,89 @@
+| Read connections | `GET /api/v2/connections` | 10 | 100 |
+| Write connections | `POST /api/v2/connections/{id}` | 10 | 100 |
+| Write device credentials | `POST /api/v2/device-credentials` | 5 | 100 | 
+| Delete device credentials | `DELETE /api/v2/device-credentials/{id}` | 5 | 100 |
```
Only applies for `Developer` and `Developer Pro` subscriptions, please create a separate section.
The remaining rate limits applies for all subscriptions.

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,89 @@
+| Verify custom domain | `POST /api/v2/custom-domains{id}/verify` | n/a | 5 | 
+| Send test email | `POST /api/v2/email-send-test` | 1 | n/a | 
+| Register dynamic client | `POST /oidc/register` | 5 | n/a | 
+| Create tenant | `POST /api/v2/tenants` | 5 | n/a | n/a |
```
```suggestion
```
Not public. Please remove.

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,89 @@
+| Register dynamic client | `POST /oidc/register` | 5 | n/a | 
+| Create tenant | `POST /api/v2/tenants` | 5 | n/a | n/a |
+| Read connection status | `GET /api/v2/connections/{id}/status` | 15 | n/a | 
+| Create tenant invitation | `POST /api/v2/tenants/invitations` | n/a | 5 | 
```
```suggestion
```
Not public. Please remove.

tamigoodall committed:
Reviewer changes

Reviewer changes

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,93 @@
+| Verify custom domain | `POST /api/v2/custom-domains{id}/verify` | n/a | 5 | 
+| Register dynamic client | `POST /oidc/register` | 5 | n/a | 
+| Read connection status | `GET /api/v2/connections/{id}/status` | 15 | n/a | 
+| All other endpoints combined | | 10 | 150 |
```
Move this line to `Developer and Developer Pro subscription limits` table

tamigoodall replied:
Which line?

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,93 @@
+| Write device credentials | `POST /api/v2/device-credentials` | 5 | 100 | 
+| Delete device credentials | `DELETE /api/v2/device-credentials/{id}` | 5 | 100 |
+
+## All subscription limits
```
```suggestion
## Endpoint limits for all subscriptions
```

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,93 @@
+
+## Rotate signing key limits
+
+The Rotate Signing Keys endpoint, `POST /api/v2/keys/signing/rotate`, has a limit of 5 per day. 
```
This can be part of `Endpoint limits for all subscriptions` table because it also applies for all subscriptions.

tamigoodall committed:
Apply suggestions from code review

Co-authored-by: Leonardo Zanivan <leo.zanivan@auth0.com>

tamigoodall committed:
Leo's changes

Leo's changes

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,90 @@
+| Read users | `GET /api/v2/users` | 40 | 500 |
+| Write users | `POST /api/v2/users` | 20 | 200 |
+| Read logs | `GET /api/v2/logs` | 10 | 100 |
+| Write logs | `GET /api/v2/user/{id}/logs` | 10 | 200|
```
```suggestion
```
Write logs limit does not exist.

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,90 @@
+
+## Developer and Developer Pro subscription limits
+
+| Endpoint | Path | Rate Limit (per second) | Rate Limit (per minute) |
```
```suggestion
| Endpoint Group | Path | Rate Limit (per second) | Rate Limit (per minute) |
```

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,90 @@
+| Endpoint | Path | Rate Limit (per second) | Rate Limit (per minute) |
+| - | - | - | - |
+| Read users | `GET /api/v2/users` | 40 | 500 |
+| Write users | `POST /api/v2/users` | 20 | 200 |
```
What we can do here? Is it no a single endpoint, but multiple paths for each group....
This is also dynamic, we can add more endpoints and do not want to miss docs updates.
Maybe we can just add one example per group?

tamigoodall replied:
I see what you're saying. I think you mentioned this in a previously resolved comment. Do we need the actual endpoint path at all? Should I just review that whole column maybe? I'm not sure I can decide. We might need a third-opinion.

panga replied:
do you think we need to add all paths for each group there? If not, please remove the column.
e.g: Read users
GET /api/v2/users
GET /api/v2/users-by-email
GET /api/v2/users/{id}

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,90 @@
+| Read connections | `GET /api/v2/connections` | 10 | 100 |
+| Write connections | `POST /api/v2/connections/{id}` | 10 | 100 |
+| Write device credentials | `POST /api/v2/device-credentials` | 5 | 100 | 
+| Delete device credentials | `DELETE /api/v2/device-credentials/{id}` | 5 | 100 |
```
```suggestion
```
This was renamed to Write device credentials.
See: https://docs.google.com/document/d/1Wr6k_KgmeaoWj5gBz7jl9BRzsgE-MzVwI_MttAOLf6s/edit#

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,90 @@
+| Read logs | `GET /api/v2/logs` | 10 | 100 |
+| Write logs | `GET /api/v2/user/{id}/logs` | 10 | 200|
+| Read clients | `GET /api/v2/clients` | 5 | 100 |
+| Write clients | `POST /api/v2/clients/{id}` | 5 | 100 |
```
```suggestion
```
Write clients limit does not exist.

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,90 @@
+| Read clients | `GET /api/v2/clients` | 5 | 100 |
+| Write clients | `POST /api/v2/clients/{id}` | 5 | 100 |
+| Read connections | `GET /api/v2/connections` | 10 | 100 |
+| Write connections | `POST /api/v2/connections/{id}` | 10 | 100 |
```
```suggestion
```
Write connections limit does not exist.

tamigoodall committed:
Merge branch 'master' into entity-rate-limits-updates

tamigoodall committed:
Apply suggestions from code review

Co-authored-by: Leonardo Zanivan <leo.zanivan@auth0.com>

tamigoodall committed:
Update rate-limits-mgmt-api.md

Added the rest of the endpoints

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -53,11 +53,23 @@ The following Auth0 Management API endpoints return rate limit-related headers.
 | Write users | `POST /api/v2/users` | 20 | 200 |
+| | `POST /api/v2/users/{id}/identities` | | |
+| | `PATCH /api/v2/users/{id}` | | |
+| | `DELETE /api/v2/users/{id}/users` | | |
```
```suggestion
| | `DELETE /api/v2/connections/{id}/users` | | |
```

tamigoodall committed:
Update articles/policies/rate-limits-mgmt-api.md

Co-authored-by: Leonardo Zanivan <leo.zanivan@auth0.com>

tamigoodall committed:
Update legacy-rate-limits.md

title fix

tamigoodall committed:
Update entity-limits.md

typo fix

tamigoodall committed:
Update entity-limits.md

Typo fix

tamigoodall committed:
Minor edits

Minor edits

panga commented on articles/policies/rate-limits-mgmt-api.md:
```diff
@@ -0,0 +1,100 @@
+---
+# Management API Endpoint Rate Limits
+
+**This policy is effective May 19, 2020. **
```
```suggestion
**This policy is effective May 19, 2020.**
```

Missing format.

tamigoodall replied:
Done

tamigoodall committed:
Update rate-limits-mgmt-api.md

bold message

santiagoaguiar commented on articles/policies/legacy-rate-limits.md:
```diff
@@ -0,0 +1,173 @@
+
+**This policy is effective for all paid and free subscriptions made before May 19, 2020.** 
+
+::: warning
```
Can we add the same sort of warning for Entity Limits policy? 

The policy will be enabled for self service and free subscriptions created on or after May 19, and for all self service and free subscriptions starting on June 18.

tamigoodall replied:
I added the note

tamigoodall commented:
Wait to merge until May 19.

tamigoodall committed:
Added note to Entity Limits

Added note to Entity Limits
Fixed title capitalization

santiagoaguiar commented on articles/policies/entity-limits.md:
```diff
@@ -0,0 +1,50 @@
+---
+# Entity Limit Policy
+
+**This policy is effective for all paid and free subscriptions made before May 19, 2020.** 
```
```suggestion
::: note
This policy is effective for all Developer, Developer Pro, and free subscriptions made on or after May 19, 2020. Starting on **June 18, 2020**, the policy will apply to all Developer, Developer Pro, and free subscriptions.
:::
```

tamigoodall committed:
Apply suggestions from code review

Co-authored-by: Santiago Aguiar <santiago.aguiar@auth0.com>
